src/index.ts

```diff
@@ -145,6 +143,7 @@ export interface ApiDataConfigAfterProps {
 export interface ApiDataBinding<T> {
     data?: T;
     request: ApiDataRequest;
-    perform: (params: EndpointParams, body: string | FormData | any) => ThunkAction<{}, { apiData: ApiDataState; }, void, Action>;
-    invalidateCache: (params: EndpointParams, body: string | FormData | any) => ThunkAction<{}, { apiData: ApiDataState; }, void, Action>;
+    perform: (params: EndpointParams, body: string | FormData | any) => Promise<ApiDataBinding<any>>;
```

Comment:
This should be updated: the params and body should be optional and the type of body does not match the actual function types. `any` should suffice.